App/Containers/Contact.tsx

```diff
@@ -119,21 +157,35 @@ const mapStateToProps = (state: RootState, ownProps: NavigationScreenProps<NavPr
   const removing = Object.keys(state.contacts.removingContacts).indexOf(address) > -1
   // Check if this contact is currently being added
   const adding = Object.keys(state.contacts.addingContacts).indexOf(address) > -1
+  const doesDirectMessageThreadExist = directMessageThreadExists(state, address)
   return {
     displayName: username ? username : address.substring(0, 12),
     threadThumbs: getThreadThumbs(state, address, 'name'),
     isContact,
     removing,
-    adding
+    adding,
+    doesDirectMessageThreadExist,
+    navigateToDirectMessageThread: () => {
+      const directMessageThread = getDirectMessageThread(state, address) as ThreadData
```

Comment:
This cast using `as ThreadData` is dangerous. The return value can actually be `undefined`. I realize there is other logic in this component that would prevent this code from being called unless there is a direct message thread, but it's better to just respect the types not make any assumptions. This is like a bug waiting to happen when this assumption is no longer valid.